src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,279 @@
-  align-self: center;
-  font-size: 18px;
-  letter-spacing: ${props.theme.fonts.letterSpacing.l};
-`,
+    background-color: ${styles.backgroundColor(props)};
+    border-radius: ${props.theme.spacings.xxsmall};
+    ${styles.border(props) && `border: ${styles.border(props)}`};
+  `,
 );
 
-const Button = ({ onClick, children, ...rest }) => (
-  <StyledButton onPress={onClick} {...rest}>
-    <StyledText {...rest}>{children}</StyledText>
-  </StyledButton>
-);
+const Button = (props) => {
+  const {
+    onClick,
+    children,
+    variant,
+    size,
+    disabled,
+    variantColor,
+    icon,
+    iconPosition,
+    align,
+    testID,
+  } = props;
+  const theme = useContext(ThemeContext);
+  return (
+    <Flex flexDirection="row" alignItems="center" alignSelf={styles.alignSelf(props)}>
+      <Space padding={styles.padding({ ...props, theme })}>
+        <StyledButton
+          onPress={onClick}
+          variantColor={variantColor}
+          disabled={disabled}
+          size={size}
+          variant={variant}
+          activeOpacity={1}
+          underlayColor={styles.underlayColor({ ...props, theme })}
+          align={align}
+          {...automation(testID)}
+        >
+          <>
+            {icon && iconPosition === 'left' && (
+              <Icon
+                name={icon}
+                size={styles.iconSize(props)}
+                fill={styles.fontColor({ ...props, theme })}
+              />
+            )}
+            {icon && iconPosition === 'left' && children && (
+              <Space margin={styles.spaceBetween(props)}>
+                <View />
+              </Space>
+            )}
+            <Text
+              color={styles.fontColor(props)}
+              size={styles.fontSize(props)}
+              align="center"
+              _weight="bold"
+              _letterSpacing={styles.letterSpacing(props)}
+              _lineHeight={styles.lineHeight(props)}
+            >
+              {size === 'xsmall' ? children && children.toUpperCase() : children}
+            </Text>
+            {icon && iconPosition === 'right' && children && (
+              <Space margin={styles.spaceBetween(props)}>
+                <View />
+              </Space>
+            )}
+            {icon && iconPosition === 'right' && (
+              <Icon
+                name={icon}
+                size={styles.iconSize(props)}
+                fill={styles.fontColor({ ...props, theme })}
+              />
+            )}
+          </>
+        </StyledButton>
+      </Space>
+    </Flex>
+  );
+};
 
 Button.propTypes = {
-  onClick: PropTypes.func.isRequired,
-  children: PropTypes.node.isRequired,
+  onClick: PropTypes.func,
+  children: PropTypes.string,
+  variant: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
+  variantColor: PropTypes.oneOf(getVariantColorKeys()),
+  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
+  align: PropTypes.oneOf(['left', 'center', 'right']),
```

Comment:
`textAlign`

Reply:
this for aligning the button, the text is always aligned center.